src/json/serialization.cr

```diff
@@ -134,13 +134,13 @@ module JSON
       # Define a `new` directly in the included type,
       # so it overloads well with other possible initializes
 
-      def self.new(pull : ::JSON::PullParser)
-        new_from_json_pull_parser(pull)
+      def self.new(pull : ::JSON::PullParser, **args)
```

Comment:
_args_ is typically used with single splats, I'd suggest sticking with _opts_ instead.

```suggestion
      def self.new(pull : ::JSON::PullParser, **opts)
```

ditto elsewhere